functions/callback-functions.md

```diff
@@ -0,0 +1,214 @@
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+  cleanupBehavior();
+}
+
+bakeCake(cleanCountertop);
+```
+<!-- prettier-ignore-end -->
+
+When we ultimately _invoke_ the `bakeCake` function with `bakeCake()`, we need to pass in an argument.
+
+<!-- prettier-ignore-start -->
+```js
+bakeCake(cleanCountertop);
+```
+<!-- prettier-ignore-end -->
+
+We're passing in `cleanCountertop`, the variable that refers to our cleaning function. We're specifically _not_ invoking the function. If we were _invoking_ it, we would use `cleanCountertop()`.
+
+Reviewing our `bakeCake` function, we can see that we invoke our `cleanupBehavior` function with `cleanupBehavior()`, without knowing if it's cleaning the countertop, cleaning the sink, cleaning the stove-top, or cleaning the bathroom!
+
+<!-- prettier-ignore-start -->
+```js
+const bakeCake = function (cleanupBehavior) {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+  cleanupBehavior();
+}
+```
+<!-- prettier-ignore-end -->
+
+We call `cleanupBehavior` a _callback function_ inside `bakeCake`. It refers to the expectation that `bakeCake` should _call back_ a function... the function that was passed in, when `bakeCake` was invoked in the first place.
+
+## Revisiting Calculator
+
+Let's consider how we can build our calculator function using callbacks. This process will be very similar to our `bakeCake` example, but we'll be mindful about how arguments are passed in.
+
+Consider this code:
+
+<!-- prettier-ignore-start -->
+```js
+const multiply = function (factor_a, factor_b) {
+  console.log('Now we\'re in multiply~~');
+  return factor_a * factor_b;
+}
+
+const calculate = function (a, b, operate) {
+  console.log('We\'re inside the calculate function!');
+  return operate(a, b);
+}
+
+const result = calculate(7, 6, multiply);
+console.log('Our final result from calculate is:', result);
+```
+<!-- prettier-ignore-end -->
+
+After running this code, we'll see this output:
+
+```
+We're inside the calculate function!
+Now we're in multiply~~
+Our final result from calculate is: 42
+```
+
+### Practice Tracing Code
+
+Let's practice tracing code, going through it and following the flow of execution. The first several lines of this snippet define functions. Our first moment of action is the line `const result = calculate(7, 6, multiply);`.
+
+We invoke our `calculate` function when we call it with parentheses `()`, in the expression `calculate(7, 6, multiply)`. Here, we're passing in the `multiply` function, without invoking it.
+
+When we enter our `calculate` function, we have these values:
+
+| Local variable | Value      |
+| -------------- | ---------- |
+| `a`            | `7`        |
+| `b`            | `6`        |
+| `operate`      | `multiply` |
+
+Our `calculate` function eventually invokes `operate` with the expression `operate(a, b)`. When we run `operate(a, b)`, we're invoking the `multiply` function with those arguments.
```

Comment:
consider including a very explicit line about how `operate` and `multiply` both refer to the same function.
```suggestion
Our `calculate` function eventually invokes `operate` with the expression `operate(a, b)`. When we run `operate(a, b)`, we're invoking the `multiply` function with those arguments. `operate` is a variable that currently holds the same function reference as `multiply`, so when we invoke `operate`, it's the same as invoking `multiply`.
```